Fix typo and simplify autoload paths intro (#37227)
rails/rails: guides/source/autoloading_and_reloading_constants.md

Autoload paths
--------------

> _Autoload paths_ is a list of application directories whose contents will be autoloaded. For example, `app/models`. Such directories represent the root namespace: `Object`.

Reviewer: Closer to the original, and clearer that we're defining a concept:

```suggestion
We refer to the list of application directories whose contents are to be autoloaded as _autoload paths_. For example, `app/models`. Such directories represent the root namespace: `Object`.
```

I think "are to be" was better; "will" is too definite, given that constants won't be autoloaded until they're used.

Can you add `[ci skip]` to your commit message to avoid running CI for a documentation-only change? Thanks!

Reviewer 2: Thank you for your feedback. Good point. And will also add `[ci skip]` to messages of documentation-only commits in future
